<launch>
    <param name="/rover/port" value="/dev/ttyRoverFTDI" type="str" />

    <param name="/rover/axleLength" value="0.102046966" type="double" />

    <param name="/rover/maxSpeed" value="0.75" type="double" />

    <param name="/rover/left/proportional" value="0.6" type="double" />
    <param name="/rover/left/integral" value="0.2" type="double" />
    <param name="/rover/left/derivative" value="0.05" type="double" />
    <param name="/rover/left/conversion_factor" value="0.004" type="double" />

    <param name="/rover/right/proportional" value="0.6" type="double" />
    <param name="/rover/right/integral" value="0.2" type="double" />
    <param name="/rover/right/derivative" value="0.05" type="double" />
    <param name="/rover/right/conversion_factor" value="0.004" type="double" />

    <param name="/rover/gyro_bias" value="0.0" />
    <param name="/rover/gyro_correction" value="0.940015506" />

    <param name="/rover/odom_frame_id" value="/odom" />
    <param name="/rover/publish_tf" value="false" />

    <node pkg="rover" name="roverBot" type="rover_node" />

    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
        <param name="output_frame" value="odom"/>
        <param name="sensor_timeout" value="1.0" />
        <param name="odom_used" value="true" />
        <param name="imu_used" value="false" />
        <param name="vo_used" value="false" />
        <param name="self_diagnose" value="true" />
        <param name="publish_tf" value="true" />

        <remap from="imu_data" to="imu/data" />
    </node>

</launch>
